src/polygon2d.jl

```diff
@@ -0,0 +1,30 @@
+"""
+```
+draw_polygon_perimeter!(img, vertices)
+draw_polygon_perimeter!(img, vertices, color)
+img2=draw_polygon_perimeter(img, vertices)
+img2=draw_polygon_perimeter(img, vertices, color)
+```
+Draws the perimeter of the polygon formed by input vertices(Array of CartesianIndex).
+"""
+
+draw_polygon_perimeter{T<:Colorant}(img::AbstractArray{T, 2}, args...) = draw_polygon_perimeter!(copy(img), args...)
+
+draw_polygon_perimeter!{T<:Colorant}(img::AbstractArray{T, 2}, vertices::Array{CartesianIndex{2},1})=draw_polygon_perimeter!(img::AbstractArray{T, 2}, vertices::Array{CartesianIndex{2},1}, one(T))
+
+function draw_polygon_perimeter!{T<:Colorant}(img::AbstractArray{T, 2}, vertices::Array{CartesianIndex{2},1}, color::T)
+
+    f = CartesianIndex(map(r->first(r)-1, indices(img)))
+    l = CartesianIndex(map(r->last(r), indices(img)))
+    for vertex in vertices
+        if min(f, vertex)!=f || max(l, vertex)!=l
+            println(vertex)
+            error("Polygon coordinates out of range.")
+        end
+    end
```

Comment:
Currently `line!` effectively checks for this because `img[x0, y0] = color` is not inside of an `@inbounds`. However, compared to checking each pixel, checking the endpoints is much more efficient. So it would be better to check this way and add `@inbounds` to the loop of `bresenham`. I think that would also work for `xiaolin_wu` but I'd want someone to take a closer look to be sure that roundoff error wouldn't cause problems.